nautobot-app/{{ cookiecutter.project_slug }}/tasks.py

```diff
@@ -673,30 +673,34 @@ def autoformat(context):
 
 @task(
     help={
-        "action": "Available values are `['lint', 'format']`. Can be used multiple times. (default: `['lint']`)",
+        "action": "Available values are `['lint', 'format']`. Can be used multiple times. (default: `['lint', 'format']`)",
+        "target": "File or directory to inspect, repeatable (default: all files in the project will be inspected)",
         "fix": "Automatically fix selected actions. May not be able to fix all issues found. (default: False)",
-        "output_format": "See https://docs.astral.sh/ruff/settings/#output-format for details. (default: `full`)",
+        "output_format": "See https://docs.astral.sh/ruff/settings/#output-format for details. (default: `concise`)",
     },
-    iterable=["action"],
+    iterable=["action", "target"],
 )
-def ruff(context, action=["lint"], fix=False, output_format="full"):
+def ruff(context, action=None, target=None, fix=False, output_format="concise"):
     """Run ruff to perform code formatting and/or linting."""
     if not action:
-        action = ["lint"]
+        action = ["lint", "format"]
+    if not target:
+        target = ["."]
```

Comment:
What's the case for doing if statements instead of putting the default value for the parameters in the function signature?

Reply:
To avoid this issue: https://pylint.pycqa.org/en/latest/user_guide/messages/warning/dangerous-default-value.html

Reply:
To be clear, it's not an issue here because we're not modifying those kwargs in the function but I think it's a good habit to avoid using mutable defaults in kwargs.